ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainActivity.kt

```diff
@@ -1,25 +1,68 @@
 package ai.nnstreamer.ml.inference.offloading
 
 import ai.nnstreamer.ml.inference.offloading.ui.theme.NnstreamerandroidTheme
+import android.annotation.SuppressLint
 import android.content.ComponentName
 import android.content.Context
 import android.content.Intent
 import android.content.ServiceConnection
 import android.os.Bundle
 import android.os.IBinder
+import android.util.Log
+import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
 import android.widget.Button
+import android.widget.ListView
 import android.widget.TextView
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.ui.Modifier
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
+
+data class ModelInfo(
+    val name: String,
+    val filter: String
+)
 
 class MainActivity : ComponentActivity() {
+    private val TAG = "MainActivity"
     private var mService: MainService? = null
 
+    inner class ModelAdapter(context: Context, resource: Int, objects: List<ModelInfo>) :
+        ArrayAdapter<ModelInfo>(context, resource, objects) {
+
+        @SuppressLint("ViewHolder")
```

Comment:
Independent models layout seems cool👍
BTW, how about use `RecyclerView` + ViewHolder rather than `ListView`? I think that `ListView` is quite old and less optimized for many items. This `SuppressLint` line could be removed too.

Reply:
Thank you for the review 😊
You're right, I think using `RecyclerView` is better :)
Updated it to use ViewHolder pattern